spring-data-neo4j-rx/src/main/java/org/neo4j/springframework/data/core/NestedRelationshipContext.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2019 "Neo4j,"
+ * Neo4j Sweden AB [https://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.springframework.data.core;
+
+import org.jetbrains.annotations.NotNull;
+import org.neo4j.springframework.data.core.mapping.Neo4jPersistentEntity;
+import org.neo4j.springframework.data.core.mapping.Neo4jPersistentProperty;
+import org.neo4j.springframework.data.core.schema.RelationshipDescription;
+import org.springframework.data.mapping.Association;
+import org.springframework.data.mapping.PersistentPropertyAccessor;
+
+/**
+ * @author Philipp Tölle
+ * @since 1.0
+ */
+final class NestedRelationshipContext {
+	private final Neo4jPersistentProperty inverse;
+	private final Object value;
+	private final RelationshipDescription relationship;
+	private final Class<?> associationTargetType;
+	private final boolean inverseValueDoesNotExists;
```

Comment:
Please use something without a negation like `inverseValue[Is]Null`/`Empty`.. or suggest something.
This also implies renaming the getter method and the variable in the `of` method.